src/hb-ot-layout-common.hh

```diff
@@ -1574,15 +1574,25 @@ static void ClassDef_remap_and_serialize (hb_serialize_context_t *c,
                                           const hb_set_t &glyphset,
                                           const hb_map_t &gid_klass_map,
                                           hb_sorted_vector_t<HBGlyphID> glyphs,
-                                          hb_sorted_vector_t<unsigned> klasses,
+                                          const hb_set_t &klasses,
                                           hb_map_t *klass_map /*INOUT*/)
 {
-  bool has_no_match = glyphset.get_population () > gid_klass_map.get_population ();
+  if (!klass_map)
+  {
+    auto it =
+    + glyphs.iter ()
+    | hb_map_retains_sorting ([&] (const HBGlyphID& gid) -> hb_pair_t<hb_codepoint_t, unsigned>
+                              {
+                                return hb_pair ((hb_codepoint_t)gid, gid_klass_map[gid]);
+                              })
+    ;
```

Comment:
I think you can also accomplish this with a zip:

hb_zip (+ glyphs.iter (), + glyphs.iter () | hb_map (gid_klass_map))